util/src/main/java/org/datacommons/util/McfChecker.java

```diff
@@ -0,0 +1,498 @@
+package org.datacommons.util;
+
+import com.google.common.base.Charsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.datacommons.proto.Debug;
+import org.datacommons.proto.Mcf;
+
+// Checks common types of nodes on naming and schema requirements.
+//
+// TODO: Add more column information
+// TODO: Pass in associated SV nodes to validate SVObs better.
+public class McfChecker {
+  private final int MAX_DCID_LENGTH = 256;
+  private final List<String> PROPS_ONLY_IN_PROP =
+      List.of(Vocabulary.DOMAIN_INCLUDES, Vocabulary.RANGE_INCLUDES, Vocabulary.SUB_PROPERTY_OF);
+  private final List<String> PROPS_ONLY_IN_CLASS = List.of(Vocabulary.SUB_CLASS_OF);
+  private final Set<String> CLASS_REFS_IN_CLASS =
+      Set.of(Vocabulary.NAME, Vocabulary.LABEL, Vocabulary.DCID, Vocabulary.SUB_CLASS_OF);
+  private final Set<String> CLASS_REFS_IN_PROP =
+      Set.of(Vocabulary.DOMAIN_INCLUDES, Vocabulary.RANGE_INCLUDES);
+  private final Set<String> PROP_REFS_IN_PROP =
+      Set.of(Vocabulary.NAME, Vocabulary.LABEL, Vocabulary.DCID, Vocabulary.SUB_PROPERTY_OF);
+  private Mcf.McfGraph graph;
+  private LogWrapper logCtx;
+  private Set<String> columns; // Relevant only when graph.type() == TEMPLATE_MCF
+  boolean foundFailure = false;
+
+  // Argument |graph| may be Instance or Template MCF.
+  public McfChecker(Mcf.McfGraph graph, LogWrapper logCtx) {
+    this.graph = graph;
```

Comment:
I found this.graph is not really used in the class. I wonder if all the check methods here could be static and just take the node or graph as argument, so the client can just do the check without constructing a checker object?

McfChecker.check(g, logCtx)

Reply:
Good idea!